src/foxops/models/incarnation.py

```diff
@@ -30,5 +31,10 @@ class Config:
 class IncarnationWithDetails(IncarnationBasic):
     status: ReconciliationStatus
 
+    template_repository: str | None
+    template_repository_version: str | None
+    template_repository_version_hash: str | None
+    template_data: dict[str, TemplateDataValue] | None
```

Comment:
I'm wondering if it would be better to wrap these fields in a `state` field -> e.g. responding the `IncarnationState` ...

Reply:
any hard feelings on this? I.e. why you would greatly prefer that approach?

Mostly because either all are None or everything is set?

Reply:
> any hard feelings on this? I.e. why you would greatly prefer that approach?

Absolutely not 😊 

> Mostly because either all are None or everything is set?

Yes, and the separation of what belongs / comes from where.

Reply:
ok... I just looked at the code again, but would propose to keep it as is for now. Just to also not increase coupling between the API responses and internal fengine models (which as of now are dataclasses).